packages/playwright-core/src/server/bidi/bidiFirefox.ts

```diff
@@ -77,7 +77,25 @@ export class BidiFirefox extends BrowserType {
     return env;
   }
 
-  override attemptToGracefullyCloseBrowser(transport: ConnectionTransport): void {
+  override async attemptToGracefullyCloseBrowser(transport: ConnectionTransport): Promise<void> {
+    if (!transport.onmessage) {
+      // browser.close does not work without an active session. If there is no connection
+      // created with the transport, make sure to create a new session first.
+      transport.send({ method: 'session.new', params: {
+        capabilities: {
+          alwaysMatch: {
+            unhandledPromptBehavior: { default: 'ignore' },
+            webSocketUrl: true
```

Comment:
You probably won't need that one given that it's basically used by WebDriver classic clients to request a BiDi session. When you directly start Firefox with `--remote-debugging-port` it will automatically start WebDriver BiDi.